File: app/src/main/java/com/tencent/iotvideo/link/popup/WxSettingDialog.java
package com.tencent.iotvideo.link.popup;

import android.content.Context;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.tencent.iot.voipdemo.R;
import com.tencent.iotvideo.link.util.VoipSetting;

public class WxSettingDialog extends IosCenterStyleDialog {

    private EditText mModelIdEt;
    private EditText mSNEt;
    private EditText mSNTicketEt;
    private EditText mAppIDEt;
    private Button mConfirmBtn;

    public WxSettingDialog(Context context) {
        super(context, R.layout.popup_wx_setting_layout);
    }

    @Override
    public void initView() {
        mModelIdEt = view.findViewById(R.id.et_voip_model_id);
        mSNEt = view.findViewById(R.id.et_voip_sn);
        mSNTicketEt = view.findViewById(R.id.et_voip_sn_ticket);
        mAppIDEt = view.findViewById(R.id.et_voip_wx_app_id);
        mConfirmBtn = view.findViewById(R.id.btn_confirm);

        mModelIdEt.setText(VoipSetting.getInstance(getContext()).modelId);
        mSNEt.setText(VoipSetting.getInstance(getContext()).sn);
        mSNTicketEt.setText(VoipSetting.getInstance(getContext()).snTicket);
        mAppIDEt.setText(VoipSetting.getInstance(getContext()).appId);

        mConfirmBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (!checkWxAppInfo()) return;
                saveWxAppInfo();
                dismiss();
                if (onDismisListener != null) {
                    onDismisListener.onDismised();
                }
            }
        });
    }

    private boolean checkWxAppInfo() {
        String modelId = mModelIdEt.getText().toString();
        String sn = mSNEt.getText().toString();
        String snTicket = mSNTicketEt.getText().toString();
        String appId = mAppIDEt.getText().toString();
        if (modelId.isEmpty() || sn.isEmpty() || snTicket.isEmpty() || appId.isEmpty()) {
            Toast.makeText(getContext(), "请输入小程序信息！", Toast.LENGTH_LONG).show();
            return false;
        }
        return true;
    }

    private void saveWxAppInfo() {
        VoipSetting.getInstance(getContext()).setModelId(mModelIdEt.getText().toString());
        VoipSetting.getInstance(getContext()).setSn(mSNEt.getText().toString());
        VoipSetting.getInstance(getContext()).setSnTicket(mSNTicketEt.getText().toString());
        VoipSetting.getInstance(getContext()).setAppId(mAppIDEt.getText().toString());
    }

    private volatile OnDismisListener onDismisListener;

    public interface OnDismisListener {
        void onDismised();
    }

    public void setOnDismisListener(OnDismisListener onDismisListener) {
        this.onDismisListener = onDismisListener;
    }
}

File: app/src/main/java/com/example/ivdemo/popup/WxSettingDialog.kt
package com.example.ivdemo.popup

import android.content.Context
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.Toast
import androidx.fragment.app.FragmentManager
import com.tencent.iot.twcall.databinding.PopupWxSettingLayoutBinding
import com.tencent.iotvideo.link.util.DeviceSetting
import com.tencent.iotvideo.link.util.updateOperate

class WxSettingDialog(context: Context) :
    IosCenterStyleDialog<PopupWxSettingLayoutBinding>(context) {
    private val deviceSetting by lazy { DeviceSetting.getInstance() }
    private val textWatcher = object : TextWatcher {
        override fun beforeTextChanged(
            s: CharSequence?,
            start: Int,
            count: Int,
            after: Int
        ) {
        }

        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
            val isOperate = checkWxAppInfo()
            binding.btnConfirm.updateOperate(isOperate)
        }

        override fun afterTextChanged(s: Editable?) {
        }

    }

    override fun getViewBinding(): PopupWxSettingLayoutBinding =
        PopupWxSettingLayoutBinding.inflate(layoutInflater)

    public override fun initView() {
        with(binding) {
            if (deviceSetting.modelId.isEmpty() || deviceSetting.appId.isEmpty()) {
                btnConfirm.updateOperate(false)
            }
            etTweCallModelId.addTextChangedListener(textWatcher)
            etTweCallModelId.setText(deviceSetting.modelId)
            etTweCallWxAppId.addTextChangedListener(textWatcher)
            etTweCallWxAppId.setText(deviceSetting.appId)
            btnConfirm.setOnClickListener(View.OnClickListener {
                if (!checkWxAppInfo()) {
                    Toast.makeText(context?.applicationContext, "请输入小程序信息！", Toast.LENGTH_SHORT).show()
                    return@OnClickListener
                }
                saveWxAppInfo()
                dismiss()
                onDismiss?.invoke()
            })
        }
    }

    private fun checkWxAppInfo(): Boolean {
        val modelId = binding.etTweCallModelId.text.toString()
        val appId = binding.etTweCallWxAppId.text.toString()
        return !(modelId.isEmpty() || appId.isEmpty())
    }

    private fun saveWxAppInfo() {
        deviceSetting.modelId = binding.etTweCallModelId.text.toString()
        deviceSetting.appId = binding.etTweCallWxAppId.text.toString()
    }

    fun show(manager: FragmentManager) {
        super.show(manager, "WxSettingDialog")
    }

    @Volatile
    private var onDismiss: (() -> Unit)? = null

    fun setOnDismissListener(onDismiss: () -> Unit) {
        this.onDismiss = onDismiss
    }
}
